packages/nodes-base/nodes/Schedule/GenericFunctions.ts

```diff
@@ -50,6 +52,46 @@ export function recurrenceCheck(
 	return false;
 }
 
+export const isIrregularInterval = (interval: ScheduleInterval): boolean => {
+	if (interval.field === 'seconds' && interval.secondsInterval >= 60) return true;
+	if (interval.field === 'minutes' && interval.minutesInterval >= 60) return true;
+	if (interval.field === 'hours' && interval.hoursInterval >= 24) return true;
+	// days and weeks are handled by recurrenceCheck so we don't consider them irregular
+	return false;
+};
+
+export const intervalToDuration = (interval: ScheduleInterval): Duration => {
+	switch (interval.field) {
+		case 'seconds':
+			return moment.duration(interval.secondsInterval, 'seconds');
+		case 'minutes':
+			return moment.duration(interval.minutesInterval, 'minutes');
+		case 'hours':
+			return moment.duration(interval.hoursInterval, 'hours');
+		case 'days':
+			return moment.duration(interval.daysInterval, 'days');
+		case 'weeks':
+			return moment.duration(interval.weeksInterval, 'weeks');
+		case 'months':
+			return moment.duration(interval.monthsInterval, 'months');
+		default:
+			throw new Error(`Invalid interval: ${interval.field}`);
+	}
+};
```

Comment:
This big switch statement can be a single property lookup as the interval field naming is consistent.